l10n_it_account_stamp/models/invoice.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from openerp import fields, api, models, exceptions
+from openerp.tools.translate import _
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def compute_stamps(self):
+        invoice_line_obj = self.env['account.invoice.line']
+        invoice_tax_obj = self.env['account.invoice.tax']
+        for inv in self:
+            stamp_product_id = self.env.user.with_context(
+                    lang=inv.partner_id.lang).company_id.tax_stamp_product_id
+            if not stamp_product_id:
+                raise exceptions.Warning(
+                    _('Missing tax stamp product in company settings!')
+                )
+            for l in inv.invoice_line:
+                if l.product_id and l.product_id.is_stamp:
+                    l.unlink()
+            taxes = invoice_tax_obj.compute(inv)
+            tax_base_amounts = {}
+            for key in taxes.keys():
+                tax_base_amounts[key[1]] = tax_base_amounts.get(
+                    key[1], 0.0) + taxes[key]['base_amount']
+            total_tax_base = 0.0
+            for tax_code_id in tax_base_amounts.keys():
+                if tax_code_id in stamp_product_id.stamp_apply_tax_ids.mapped(
+                        'base_code_id.id'):
+                    total_tax_base += tax_base_amounts[tax_code_id]
+            taxes = stamp_product_id.taxes_id
+            if inv.type in ('in_invoice', 'in_refund'):
+                total_tax_base = total_tax_base * -1.0
+                taxes = stamp_product_id.supplier_taxes_id
+            if inv.fiscal_position:
+                taxes_ids = inv.fiscal_position.map_tax(taxes)
```

Comment:
Se in fattura ci fosse una posizione fiscale, a questo punto (in `action_move_create`) le imposte non sarebbero già state mappate?
In quale caso servirebbe questo `map_tax`?